lib/AArch64Base.ml

```diff
@@ -441,6 +441,7 @@ type 'k kinstruction =
   | I_STOPBH of  atomic_op * bh * w_type  * reg * reg
 (* Operations *)
   | I_MOV of variant * reg * 'k kr
+  | I_MOVZ of variant * reg * 'k kr * 'k s
```

Comment:
As I understand it the instruction MOVZ always takes an immediate constant as second argument. As a consequence, the following definition would be more logical.
```suggestion
  | I_MOVZ of variant * reg * 'k * 'k s
```
Of course, such a change will entail many other changes.Those should quite easy, with typechecker guidance.